File: Modulo10/UniversidadeDepartamento/Departamento.cpp
#include "Departamento.h"
#include "Universidade.h"

#include <iostream>
using namespace std;

Departamento::Departamento(string nome, string sigla, Universidade* universidade) :
nome(nome), sigla(sigla), universidade(universidade) {
    cout << "[Construindo departamento " << sigla << "]" << endl;
}

Departamento::~Departamento() {
    cout << "[Destruindo departamento " << sigla << "]" << endl;
}

string Departamento::getNome() const {
    return nome;
}

void Departamento::setNome(string nome) {
    this->nome = nome;
}

string Departamento::getSigla() const {
    return sigla;
}

void Departamento::setSigla(string sigla) {
    this->sigla = sigla;
}

Universidade* Departamento::getUniversidade() const {
    return universidade;
}

void Departamento::setUniversidade(Universidade* universidade) {
    this->universidade = universidade;
}

void Departamento::imprime() const {
    cout << "Departamento " << nome << " de sigla " << sigla;
    cout << " (" << universidade->getNome() << ")" << endl;
}

File: Modulo07/UniversidadeDepartamento-Java/Departamento.java

public class Departamento {
    private String nome;
    private String sigla;
    private Universidade universidade;

    public Departamento(String nome, String sigla, Universidade universidade) {
        this.nome = nome;
        this.sigla = sigla;
        this.universidade = universidade;
        System.out.println("[Construindo departamento " + sigla + "]");
    }

    public String getNome() {
        return this.nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getSigla() {
        return this.sigla;
    }

    public void setSigla(String sigla) {
        this.sigla = sigla;
    }

    public Universidade getUniversidade() {
        return this.universidade;
    }

    public void setUniversidade(Universidade universidade) {
        this.universidade = universidade;
    }

    public void imprime() {
        System.out.print("Departamento " + nome + " de sigla " + sigla);
        System.out.println(" (" + universidade.getNome() + ")");
    }
}
